diff --git a/admin_menu.inc b/admin_menu.inc
index 2eab789..e6169df 100644
--- a/admin_menu.inc
+++ b/admin_menu.inc
@@ -204,7 +204,7 @@ function admin_menu_merge_tree(array &$tree, array $tree_dynamic, array $expand_
         foreach ($load_functions as $index => $function) {
           if ($function) {
             if (is_array($function)) {
-              list($function,) = each($function);
+              $function = key($function);
             }
             // Add the loader function name minus "_load".
             $placeholder = '%' . substr($function, 0, -5);
