diff --git i/imageapi_gd.module w/imageapi_gd.module
index 627b15a..a8ce627 100644
--- i/imageapi_gd.module
+++ w/imageapi_gd.module
@@ -51,7 +51,6 @@ function imageapi_gd_unsharp_mask($img, $radius, $sigma, $amount, $threshold)
   $radius = abs(round($radius)); // Only integers make sense.
   if ($radius == 0) {
     return $img; imagedestroy($img);
-    break;
   }
 
   $w = imagesx($img); $h = imagesy($img);
