diff --git a/editablefields.module b/editablefields.module
index 4a30817..1dfa8b2 100755
--- a/editablefields.module
+++ b/editablefields.module
@@ -223,7 +223,7 @@ function editablefields_fallback_formatter($entity_type, $entity, $field, $insta
  */
 function editablefields_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, &$items, $display) {
   if (!$display['type'] == 'editable') {
-    continue;
+    return;
   }
 
   $field_name = $field['field_name'];
